lib/ansible/modules/cloud/amazon/aws_glacier_vault.py

```diff
@@ -0,0 +1,231 @@
+#!/usr/bin/python
+# Copyright (c) 2018 Loic BLOT <loic.blot@unix-experience.fr>
+# This module is sponsored by E.T.A.I. (www.etai.fr)
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: aws_glacier_vault
+short_description: Manage Glacier vaults in AWS
+description:
+    - Manage Glacier vaults in AWS
+version_added: "2.6"
+requirements: [ boto3 ]
+author: "Loic Blot <loic.blot@unix-experience.fr>"
+options:
+  name:
+    description:
+      - Name of the glacier vault
+    required: true
+    default: null
+  state:
+    description:
+      - Create or remove the glacier vault
+    required: false
+    default: present
+    choices: [ 'present', 'absent' ]
+  tags:
+    description:
+      - A hash/dictionary of tags to add to the new instance or to add/remove from an existing one.
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+RETURN = '''
+vault:
+    description: the vault object
+    returned: always
+    type: complex
+    contains:
+        creation_date:
+            description: The vault creation date.
+            returned: always
+            type: string
+            sample: "2018-02-13T22:10:28.590Z"
+        number_of_archives:
+            description: Number of archives currently in the vault.
+            returned: always
+            type: int
+            sample: 0
+        size_in_bytes:
+            description: Current vault size.
+            returned: always
+            type: int
+            sample: 0
+        vault_arn:
+            description: The vault AWS ARN.
+            returned: always
+            type: string
+            sample: "arn:aws:glacier:eu-west-3:123457788994:vaults/test-vault"
+        vault_name:
+            description: The vault name.
+            returned: always
+            type: string
+            sample: "test-vault"
+tags:
```

Comment:
This structure seems wrong to me. I'd rather have all the properties at the same top level.

One way to do that without having to change your implementation too much would be to 

```
module.exit_json(**vault, tags=self.list.tags(), changed=changed)
```

Reply:
if i do this i cannot do the camel conversion and push changes and tags

Reply:
You can, just put it last.
```
(python2.7) shertel-OSX:ansible shertel$ git diff
diff --git a/lib/ansible/modules/cloud/amazon/aws_glacier_vault.py b/lib/ansible/modules/cloud/amazon/aws_glacier_vault.py
index 9eb27ef79e..2db590d6d1 100644
--- a/lib/ansible/modules/cloud/amazon/aws_glacier_vault.py
+++ b/lib/ansible/modules/cloud/amazon/aws_glacier_vault.py
@@ -142,7 +142,7 @@ class GlacierVaultManager(object):
         vault = self.get_vault(self.name)

         # Vault exists, no change to do
-        self.module.exit_json(vault=camel_dict_to_snake_dict(vault), tags=self.list_tags(), changed=changed)
+        self.module.exit_json(tags=self.list_tags(), changed=changed, **camel_dict_to_snake_dict(vault))
```